force-app/main/default/classes/Application_TEST.cls

```diff
@@ -0,0 +1,151 @@
+*/
+/**
+ * @author Salesforce.org
+ * @date 2021
+ * @group Contacts
+ * @group-content ../../ApexDocContent/Contacts.htm
+ * @description Test class for logic that prevents application due date to be prior to the open date.
+ */
+@isTest
+private with sharing class Application_TEST {
+    @TestSetup
+    static void makeData() {
+        List<Account> academicAccount = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            1,
+            UTIL_Describe_API.getAcademicAccRecTypeID()
+        );
+        insert academicAccount;
+
+        Contact student = UTIL_UnitTestData_TEST.getContact();
+        insert student;
+    }
+    /**************************************************************************************************************************
+     ****************************************************** UNIT TESTS *********************************************************
+     **************************************************************************************************************************/
+
+    /********************************************************
+     * @description Should fail to insert an application with a due date prior to the open date
+     *********************************************************/
+    @isTest
+    static void shouldFailInsertWhenApplicationDueDateIsBeforeOpenDate() {
+        Account academicAccount = getAcademicAccount();
+        Contact student = getStudent();
+
+        Application__c invalidApplication = new Application__c(
+            Applicant__c = student.Id,
+            Applying_To__c = academicAccount.Id,
+            Due_Date__c = System.today().addDays(-10),
+            Open_Date__c = System.today()
+        );
+
+        Test.startTest();
+        Database.SaveResult insertResult = Database.insert(invalidApplication, false);
+        Test.stopTest();
+
+        System.assertEquals(false, insertResult.isSuccess(), 'Insert should have failed due to invalid date range');
+        System.assertEquals(
+            Label.stgErrorAppDueDate,
+            insertResult.getErrors()[0].getMessage(),
+            'Should fail with the Open/Due Date validation error message'
+        );
+    }
+
+    /********************************************************
+     * @description Should fail to update an application when setting the due date prior to the open date
+     *********************************************************/
+    @isTest
+    static void shouldFailUpdateWhenApplicationDueDateIsBeforeOpenDate() {
+        Account academicAccount = getAcademicAccount();
+        Contact student = getStudent();
+
+        Application__c openEndedApplication = new Application__c(
+            Applicant__c = student.Id,
+            Applying_To__c = academicAccount.Id,
+            Open_Date__c = System.today()
+        );
+        insert openEndedApplication;
+
+        Test.startTest();
+        openEndedApplication.Due_Date__c = System.today().addDays(-10);
```

Comment:
This is completely a nitpick and is not required. Given that the test block is used to create a separate set of limits and we can run performance metrics on that, anything that is not directly the test likely should remain outside the block.

```suggestion
        openEndedApplication.Due_Date__c = System.today().addDays(-10);

        Test.startTest();
```